proxyd/backend.go

```diff
@@ -845,6 +891,17 @@ func (bg *BackendGroup) ProxyWS(ctx context.Context, clientConn *websocket.Conn,
 	return nil, ErrNoBackends
 }
 
+func (bg *BackendGroup) orderedBackendsForRequest() []*Backend {
+	backends := bg.Backends
+	if bg.Consensus != nil {
+		backends = bg.loadBalancedConsensusGroup()
+	} else if bg.WeightedRouting {
+		choice := bg.weightedChooser.Pick()
+		backends = moveBackendToStart(choice, backends)
+	}
+	return backends
+}
+
```

Comment:
consensus and weight routing shouldn't be mutually exclusive: we should be able to turn on consensus awareness + weighted routing at same time

for classic:
ideally we would do the weighted shuffling using the same list of backends as input

for consensus:
ideally we would do the weighted shuffling using the healthy list of the backends as input; then append the list of degraded backends (which does not need to be shuffled at all)

Reply:
In the PR, they're not mutually exclusive `loadBalancedConsensusGroup` has logic in it to handle the case when weighted routing is enabled.